### PR TITLE
EWPP-672: Contextual filters selection.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+      $values = $configuration['contextual_filters'] ?? [];
+      static::setCurrentValues($form_state, $list_source, $values);
+      return;
+    }
+
+    static::setCurrentValues($form_state, $list_source, []);
+  }
+
+  /**
+   * Ajax request handler for setting a default value for a filters.
```
Comment (imanoleguskiza) [inline modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php:232]: Probably something like "Ajax request handler for setting an operator for a contextual filter." would be better.

File: modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+   *   The form element.
+   */
+  public function setOperatorAjax(array &$form, FormStateInterface $form_state): array {
+    $triggering_element = $form_state->getTriggeringElement();
+    $element = NestedArray::getValue($form, array_slice($triggering_element['#array_parents'], 0, -4));
+    return $element['wrapper'];
+  }
+
+  /**
+   * Submit callback for setting a default value for a filters.
```
Comment (imanoleguskiza) [inline modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php:249]: Similar to above, it should be something like: "Submit callback for setting an operator for a contextual filter."

File: modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+   *   The facet ID.
+   * @param string $filter_id
+   *   The filter ID.
+   * @param \Drupal\oe_list_pages\ListSourceInterface $list_source
+   *   The list source.
+   *
+   * @return array
+   *   The built form.
+   */
+  protected function buildEditPresetFilter(array $form, FormStateInterface $form_state, string $facet_id, string $filter_id, ListSourceInterface $list_source): array {
```
Comment (imanoleguskiza) [inline modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php:121]: buildEditContextualFilter maybe?

File: modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php [hunk 4]
```diff
@@ -0,0 +1,330 @@
...
+
+/**
+ * Builds the list page source contextual filters form elements.
+ */
+class ContextualFiltersConfigurationBuilder extends FilterConfigurationFormBuilderBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  protected function getAjaxWrapperId(array $form): string {
```
Comment (imanoleguskiza) [inline modules/oe_list_pages_link_list_source/src/ContextualFiltersConfigurationBuilder.php:22]: We could maybe make a base function for this using getFilterType to generate the id but not required.

**Review by imanoleguskiza — CHANGES_REQUESTED** [review]

**Review by imanoleguskiza — APPROVED** [review]

**Review by hernani — APPROVED** [review]

**Pull request merged**